[UT]TestRunner_Start/exec_once does not stop child at a low probility

Please note that the Consul Template issue tracker is reserved
for bug reports and enhancements. For general usage questions,
please use the Consul mailing list:

    https://groups.google.com/forum/#!forum/consul-tool

### Consul Template version

master branch until https://github.com/hashicorp/consul-template/commit/fd266d9f1724ed88681517ba55ce909024e16d30

### What happend
`TestTestSuitePidOf/TestMultipleProcesses` and `TestTestSuitePidOf/TestSingleProcess` failed.

```
00:55:02 --- FAIL: TestTestSuitePidOf (0.40s)
00:55:02     --- PASS: TestTestSuitePidOf/TestInvalidProcessName (0.00s)
00:55:02     --- FAIL: TestTestSuitePidOf/TestMultipleProcesses (0.19s)
00:55:02     	assertions.go:247: 
                          
	Error Trace:	lbmgr_test.go:99
00:55:02     		
	Error:      	"[%!s(int=7272) %!s(int=7425) %!s(int=7426) %!s(int=7427) %!s(int=7428) %!s(int=7429) %!s(int=7430) %!s(int=7431) %!s(int=7432) %!s(int=7433) %!s(int=7434) %!s(int=7435) %!s(int=7436) %!s(int=7437) %!s(int=7438)]" should have 14 item(s), but has 15
00:55:02     		
	Test:       	TestTestSuitePidOf/TestMultipleProcesses
00:55:02     --- PASS: TestTestSuitePidOf/TestNoSuchProcess (0.10s)
00:55:02     --- PASS: TestTestSuitePidOf/TestProcessNameEmpty (0.00s)
00:55:02     --- FAIL: TestTestSuitePidOf/TestSingleProcess (0.10s)
00:55:02     	assertions.go:247: 
                          
	Error Trace:	lbmgr_test.go:86
00:55:02     		
	Error:      	"[%!s(int=7272) %!s(int=7439)]" should have 1 item(s), but has 2
```

### REASON
Both of the test check whether or not the number of processes the started is as expected, but unfortunately they failed because **7272** is not killed from previous test `TestRunner_Start/exec_once`

--- a/manager/runner_test.go
+++ b/manager/runner_test.go
@@ -757,7 +757,23 @@ func TestRunner_Start(t *testing.T) {
 		case err := <-r.ErrCh:
 			t.Fatal(err)
 		case <-r.renderedCh:
-			// Just assert there is no panic
+			found := false
+			for i := 0; i < 5; i++ {
+				if found {
+					break
+				}
+
+				time.Sleep(100 * time.Millisecond)
+
+				r.childLock.RLock()
+				if r.child != nil {
+					found = true
+				}
+				r.childLock.RUnlock()
+			}
+			if !found {
+				t.Error("missing child")
+			}
 		case <-time.After(2 * time.Second):
 			t.Fatal("timeout")
 		}

